arangod/Aql/ExecutionBlockImpl.tpp

```diff
@@ -2663,6 +2721,10 @@ void ExecutionBlockImpl<Executor>::PrefetchTask::wakeupWaiter() noexcept {
   _state.store({Status::Finished, true}, std::memory_order_release);
   _lock.unlock();
 
+  if (_timeoutInWait.load(std::memory_order_relaxed)) {
+    LOG_TOPIC("62518", WARN, Logger::AQL)
+        << "PrefetchTask: notify_one happens after timeout saw FINISHED";
```

Comment:
To be nitpicky, we don't know that it was actually `Finished`, just that it wasn't `InProgress`.

And just for me so I understand this correctly. If we see this message, we're sure that we were just late to ring the bell. However, if we don't see it, but we do see `[62514]`, we still don't know - right?

Reply:
I fear you are right. The special interleaving which would be needed for this should be even more unlikely - but not impossible. I wonder how to detect this reliably?

Reply:
Removed